tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -160,41 +173,42 @@ IotHttpsResponseHandle_t newIotResponseHandle() {
 IotHttpsRequestHandle_t newIotRequestHandle() {
   IotHttpsRequestHandle_t pRequestHandle = safeMalloc(sizeof(_reqHandle_t));
   if (pRequestHandle) {
+    uint32_t len;
     pRequestHandle->pHttpsResponse = newIotResponseHandle();
     pRequestHandle->pHttpsConnection = newIotConnectionHandle();
-    pRequestHandle->pBody = malloc(sizeof(uint32_t));
-    pRequestHandle->pHeaders = safeMalloc(sizeof(pRequestHandle) + sizeof(struct _httpRequest));
+    pRequestHandle->pBody = malloc(len);
+    pRequestHandle->pHeaders = ((_reqHandle_t*)pRequestHandle)->data;
     pRequestHandle->pHeadersCur = pRequestHandle->pHeaders;
-    pRequestHandle->pHeadersEnd = safeMalloc(sizeof(pRequestHandle) + sizeof(_reqHandle_t));
+    pRequestHandle->pHeadersEnd = pRequestHandle->pHeaders + sizeof(((_reqHandle_t*)pRequestHandle)->data);
     pRequestHandle->pConnInfo = newConnectionInfo();
   }
   return pRequestHandle;
 }
-
 /* Creates a Request Info and assigns memory accordingly. */
 IotHttpsRequestInfo_t * newIotRequestInfo() {
   IotHttpsRequestInfo_t * pReqInfo = safeMalloc(sizeof(IotHttpsRequestInfo_t));
-  uint32_t bufferSize;
   if(pReqInfo) {
+    uint32_t bufferSize;
+    uint32_t hostNameLen;
     __CPROVER_assume(bufferSize >=0 && bufferSize <= requestUserBufferMinimumSize);
     pReqInfo->userBuffer.bufferLen = bufferSize;
     pReqInfo->userBuffer.pBuffer = safeMalloc(bufferSize);
+    __CPROVER_assume(hostNameLen >= 0 && hostNameLen <= IOT_HTTPS_MAX_HOST_NAME_LENGTH + 1);
+    pReqInfo->pHost = safeMalloc(hostNameLen);
   }
   return pReqInfo;
 }
 
 /* Creates a Response Info and assigns memory accordingly. */
 IotHttpsResponseInfo_t * newIotResponseInfo() {
   IotHttpsResponseInfo_t * pRespInfo = safeMalloc(sizeof(IotHttpsResponseInfo_t));
-  uint32_t bufferSize;
-  uint32_t bodySize;
   if(pRespInfo) {
-    __CPROVER_assume(bufferSize >= 0 && bufferSize <= responseUserBufferMinimumSize);
+    uint32_t bufferSize;
+    uint32_t bodySize;
     pRespInfo->userBuffer.bufferLen = bufferSize;
-    pRespInfo->userBuffer.pBuffer = safeMalloc(bufferSize);
+    pRespInfo->userBuffer.pBuffer = malloc(bufferSize);
     pRespInfo->pSyncInfo = malloc(sizeof(IotHttpsSyncInfo_t));
-    __CPROVER_assume(bodySize >= responseUserBufferMinimumSize);
-    pRespInfo->pSyncInfo->pBody = safeMalloc(bodySize);
+    pRespInfo->pSyncInfo->pBody = malloc(bodySize);
```

Comment:
Add a comment saying you are assuming these pointers cannot be null (in contrast to other pointers assigned with safeMalloc.